credentialproviderpackage/pkg/configurator/linux/linux.go

```diff
@@ -0,0 +1,223 @@
+package linux
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"syscall"
+
+	ps "github.com/mitchellh/go-ps"
+
+	"credential-provider/pkg/configurator"
+	"credential-provider/pkg/constants"
+	"credential-provider/pkg/log"
+	"credential-provider/pkg/templater"
+)
+
+//go:embed templates/credential-provider-config.yaml
+var credProviderTemplate string
+
+type linuxOS struct {
+	profile       string
+	extraArgsPath string
+	basePath      string
+	config        constants.CredentialProviderConfigOptions
+}
+
+var _ configurator.Configurator = (*linuxOS)(nil)
+
+func NewLinuxConfigurator() *linuxOS {
+	return &linuxOS{
+		profile:       "",
+		extraArgsPath: constants.MountedExtraArgs,
+		basePath:      constants.BasePath,
+	}
+}
+
+func (c *linuxOS) Initialize(config constants.CredentialProviderConfigOptions) {
+	c.config = config
+}
+
+func (c *linuxOS) UpdateAWSCredentials(sourcePath string, profile string) error {
+	c.profile = profile
+	dstPath := c.basePath + constants.CredOutFile
+
+	err := copyWithPermissons(sourcePath, dstPath, 0600)
+	return err
+}
+
+func (c *linuxOS) UpdateCredentialProvider(_ string) error {
+	// Adding to KUBELET_EXTRA_ARGS in place
+	file, err := ioutil.ReadFile(c.extraArgsPath)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(file), "\n")
+	found := false
+	for i, line := range lines {
+		if strings.HasPrefix(line, "KUBELET_EXTRA_ARGS") {
+			found = true
+			args := c.updateKubeletArguments(line)
+
+			if args != "" {
+				lines[i] = line + args + "\n"
+			}
+		}
+	}
+	if !found {
+		line := "KUBELET_EXTRA_ARGS="
+		args := c.updateKubeletArguments(line)
+		if args != "" {
+			line = line + args
+		}
+		lines = append(lines, line)
+	}
+
+	out := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(c.extraArgsPath, []byte(out), 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *linuxOS) CommitChanges() error {
+	process, err := findKubeletProcess()
+	if err != nil {
+		return err
+	}
+	err = killProcess(process)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
There are many instances of similar things across the PR. It doesn't affect functionality but the if is unnecessary.

```suggestion
	return err
```